src/workers.cc

```diff
@@ -378,14 +378,14 @@ void KafkaConsumerConsumeLoop::Execute(const ExecutionMessageBus& bus) {
       // when in consume loop mode
       // Randomise the wait time to avoid contention on different
       // slow topics
-      usleep(static_cast<int>(rand_r(&m_rand_seed) * 1000 * 1000 / RAND_MAX));
+      std::this_thread::sleep_for(std::chrono::milliseconds(static_cast<int>(rand() * 1000 / RAND_MAX)));
```

Comment:
GNU `rand()` is not thread-safe, but `rand_r(*seedp)` is. Only MSVC's `rand()` is thread-safe. Therefore it seems that `rand()` should only be used in this context if it is being compiled by MSVC, not on all platforms.